CMake error when running `build-script` with `--xcode`

<!--
This repository tracks issues related to the implementation of the Swift
compiler, standard library, runtime, and tools that provide IDE support for
Swift (e.g. code completion). If the bug relates to the implementation of a
proprietary (closed-source) Apple framework such as UIKit, SwiftUI, Combine,
etc., please report it to https://feedbackassistant.apple.com instead.
-->

**Description**
<!-- Describe clearly and concisely what the bug is. -->
Build script fails on running:
```
utils/build-script --skip-build-benchmarks --skip-ios \
   --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
   --swift-disable-dead-stripping --release-debuginfo --xcode --clean
```

This issue appears to be similar to the issue discussed in #62205

**Steps to reproduce**
<!--
Explain how to reproduce the problem (in steps if seen fit) and include either
an inline test case (preferred) or a project that reproduces it. Consider
reducing the sample to the smallest amount of code possible — a smaller test
case is easier to reason about and more appealing to сontributors.
-->

1. Checkout main: `utils/update-checkout --scheme main` (current revision: 244ca4e2426260e7b9161c2fd6534dc350983cdf)
2. Run the build script using the following command:
```
utils/build-script --skip-build-benchmarks --skip-ios \
   --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
   --swift-disable-dead-stripping --release-debuginfo --xcode --clean
```

There following errors occur, 4 times each:

```
CMake Error in lib/ASTGen/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftBasicFormat" includes non-existent path

    "/Users/whiteio/Development/swift-project/build/Xcode-RelWithDebInfoAssert/earlyswiftsyntax-macosx-arm64/lib/swift/host"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

```
CMake Error in lib/Parse/CMakeLists.txt:
  Imported target "SwiftSyntax::SwiftBasicFormat" includes non-existent path

    "/Users/whiteio/Development/swift-project/build/Xcode-RelWithDebInfoAssert/earlyswiftsyntax-macosx-arm64/lib/swift/host"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Expected behavior**

It should successfully build and create the Xcode projects.

**Environment**

swift-driver version: 1.62.15 
Apple Swift version 5.7.1 (swiftlang-5.7.1.135.3 clang-1400.0.29.51)
Target: arm64-apple-macosx12.0

Xcode 14.1
Build version 14B47b

--- a/utils/build_swift/build_swift/driver_arguments.py
+++ b/utils/build_swift/build_swift/driver_arguments.py
@@ -126,6 +126,7 @@ def _apply_default_arguments(args):
         # Building with Xcode is deprecated.
         args.skip_build = True
         args.build_early_swift_driver = False
+        args.build_early_swiftsyntax = False
 
     # --ios-all etc are not supported by open-source Swift.
     if args.ios_all:

--- a/utils/build_swift/tests/build_swift/test_driver_arguments.py
+++ b/utils/build_swift/tests/build_swift/test_driver_arguments.py
@@ -646,3 +646,4 @@ class TestDriverArgumentParser(
         self.assertEqual(namespace.cmake_generator, 'Xcode')
         self.assertTrue(namespace.skip_build)
         self.assertFalse(namespace.build_early_swift_driver)
+        self.assertFalse(namespace.build_early_swiftsyntax)

